app/ui/legacy/src/main/java/com/fsck/k9/activity/setup/DefaultAccountSettings.java

```diff
@@ -0,0 +1,22 @@
+package com.fsck.k9.activity.setup;
+
+import com.fsck.k9.mail.AuthType;
+
+public class DefaultAccountSettings implements java.io.Serializable {
```

Comment:
I think something like `BasicAccountSettings` or `InitialAccountSettings` would be a better name.

New code should be written in Kotlin. Instead of `Serializable` we can make this a `Parcelable`.

```kotlin
@Parcelize
data class DefaultAccountSettings(
    val authenticationType: AuthType,
    val email: String,
    val password: String?,
    val clientCertificateAlias: String?
) : Parcelable
```

Reply:
Good call on the name change. I've done that and changed to kotlin as suggested. 

Is the use of `parcelable` over `serializable` a speed issue? (I haven't dug into the comparison before but that's what's my googling suggested)

Potentially related question. Whats the `legacy` file structure that this is in used for? My assumption was it's for java but there is kotlin files in there too.

Reply:
> Is the use of `parcelable` over `serializable` a speed issue? (I haven't dug into the comparison before but that's what's my googling suggested)

It used to be. Not sure if that's still very relevant. Mainly, it's the canonical way of doing things on Android.

> Potentially related question. Whats the `legacy` file structure that this is in used for? My assumption was it's for java but there is kotlin files in there too.

It's part of the name of the Gradle module (`:app:ui:legacy`). Currently it contains all of the UI code. The goal is to split this up into multiple modules, one per feature (or group of closely related features). That will help with compilation avoidance, but could also be useful for custom K-9 Mail versions/forks that don't want to include certain features.